### ISSUE TITLE
Update mentions and usages of `package:lints` version to 2.0

### ISSUE THREAD

**Issue by parlough** [dart-lang/site-www#3923]
With Dart 2.17, the lints package will release a ^2.0.0 version with various new lints and corresponding fixes. 

If we mention or use the old version anywhere, with the release of 2.17 we should update those references and fix any lints as necessary.

**Comment by domesticmouse** [comment]
Is `flutter_lints` also getting a matching major version bump?

**Comment by parlough** [comment]
@domesticmouse Yep, you can find more info here: https://github.com/flutter/packages/pull/1165, but I'm not sure about the timelines on the `flutter_lints` side.

**Comment by goderbauer** [comment]
v2 of package:lints is published. package:flutter_lints will follow shortly as soon as https://github.com/flutter/packages/pull/1165#issuecomment-1071542929 is resolved.
